third_party/move/move-compiler-v2/src/pipeline/ability_checker.rs

```diff
@@ -0,0 +1,256 @@
+//! Checks for ability violations.
+
+use itertools::Itertools;
+use move_binary_format::file_format::{Ability, AbilitySet};
+use move_model::{
+    ast::TempIndex,
+    model::{FunId, FunctionEnv, Loc, ModuleId, QualifiedId, StructId, TypeParameter},
+    ty,
+    ty::Type,
+};
+use move_stackless_bytecode::{
+    function_target::{FunctionData, FunctionTarget},
+    function_target_pipeline::{FunctionTargetProcessor, FunctionTargetsHolder},
+    stackless_bytecode::{AssignKind, Bytecode, Operation},
+};
+
+// Returns the abilities of the given type.
```

Comment:
Use `///` instead of `//`

Reply:
done

Reply:
> There are couple of TODOs in live_var_processor about ability check for copy. Are those now obsolete? Or do we still need to do this? In the earlier case, lets remove those TODOs.
No. I plan to insert checks in the live_var_processor for better error messages of implicit copies.